src/main/java/org/jenkinsci/plugins/workflow/actions/ArgumentsAction.java

```diff
@@ -164,7 +162,7 @@ public String getUrlName() {
         if (internalArgs.size() == 0) {
             return Collections.emptyMap();
         }
-        HashMap<String, Object> filteredArguments = Maps.newHashMapWithExpectedSize(internalArgs.size());
+        HashMap<String, Object> filteredArguments = new HashMap(internalArgs.size());
```

Comment:
Note that there is a subtle difference between these, see the discussion of `loadFactor` in the `HashMap` documentation.

Reply:
Yes I know but looking at the context. It probably doesn't hurt...
```
        HashMap<String, Object> filteredArguments = new HashMap(internalArgs.size());
        for (Map.Entry<String, Object> entry : internalArgs.entrySet()) {
            if (entry.getValue() != null && !(entry.getValue() instanceof NotStoredReason)) {
                // TODO this is incorrect: value could be a Map/List with some nested entries that are NotStoredReason
                filteredArguments.put(entry.getKey(), entry.getValue());
            }
        }
``` 
If 25% of the internalArgs `Map` is filtered by the `if` condition this `optimisation` with size/loadfactor doesn't make any sense.
and by the way looking at the rest of the code this `Map` is only read. 
I might more using something such?:
```
        Map<String, Object>  filteredArguments = internalArgs.entrySet().stream()
            .filter(entry -> entry.getValue() != null && !(entry.getValue() instanceof NotStoredReason))
            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```